### PR TITLE
[query] Added functionality to time parsing logic

### PR DESCRIPTION
**Pull request by teddywahle**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

**Review by robskillington — APPROVED** [review]
LGTM

**Comment by robskillington** [comment]
Following packages are failing CI:
```
./src/query/graphite/graphite
./src/query/graphite/native
```

File: src/query/graphite/common/basic_functions.go [hunk 1]
```diff
@@ -181,19 +182,25 @@ func Count(ctx *Context, seriesList ts.SeriesList, renamer SeriesListRenamer) (t
...
 // ParseInterval parses an interval string and returns the corresponding duration.
-func ParseInterval(s string) (time.Duration, error) {
-	if m := reInterval.FindStringSubmatch(strings.TrimSpace(s)); len(m) != 0 {
-		amount, err := strconv.ParseInt(m[1], 10, 32)
+func ParseInterval(fullInterval string) (time.Duration, error) {
+	allIntervals := reInterval.FindAllString(fullInterval, -1)
+	output := time.Duration(0)
+	if allIntervals == nil {
+		return 0, errors.NewInvalidParamsError(fmt.Errorf("Unrecognized interval string: %s", fullInterval))
+	}
```
Comment (teddywahle) [inline src/query/graphite/common/basic_functions.go:190]:
```suggestion
	}
	
```

File: src/query/graphite/graphite/timespec.go [hunk 2]
```diff
@@ -40,13 +40,30 @@ var reAM = regexp.MustCompile(`(?i)^(([0-1]?)([0-9])am)([[:ascii:]])*$`)
...
-	"d":   time.Hour * 24,
-	"w":   time.Hour * 24 * 7,
-	"mon": time.Hour * 24 * 30,
-	"y":   time.Hour * 24 * 365,
+	"s":       time.Second,
+	"sec":     time.Second,
+	"seconds": time.Second,
+	"m":       time.Minute,
+	"min":     time.Minute,
+	"mins":     time.Minute,
```
Comment (teddywahle) [inline src/query/graphite/graphite/timespec.go:48]:
```suggestion
	"mins":    time.Minute,
```

**Review by robskillington — APPROVED** [review]
LGTM

**Pull request merged**